frontend/src/app/Catalog/CatalogView/CatalogView.tsx

```diff
@@ -130,11 +256,115 @@ const CatalogView = (props: CatalogViewProps) => {
 		});
 	};
 
-	const enrollPath = legacyId
-		? `/enrollment/0-${legacyId}-${semester.replace(' ', '-')}-all`
-		: `/enrollment`;
+	
 
-	const gradePath = legacyId ? `/grades/0-${legacyId}-all-all` : `/grades`;
+
+	const [hoveredClass, setHoveredClass] = useState<any>(false);
+	const [updateMobileHover, setUpdateMobileHover] = useState<any>(true);
+	
+
+	function update(course:any, grade:any) {
+		const gradesData = (store.getState() as any).grade.gradesData;
+		if (course && gradesData && gradesData.length > 0) {
+			const selectedGrades = gradesData.filter((c:any) => course.id === c.id)[0];
+			const hoverTotal = {
+				...course,
+				...selectedGrades,
+				hoverGrade: grade
+			};
+
+			setHoveredClass(hoverTotal);
+		}
+	}
+
+	// Handler function for updating GradesInfoCard on hover with single course
+	function updateGraphHover(data:any) {
```

Comment:
Lets learn what these types are or atleast  attempt to understand the shape of the data so we can get better type safety. There shouldn't be any `any` types anywhere.

Reply:
I'm going to leave this one as this came with the old Graphs component. I just copy pasted